complex/ft_test.c

```diff
@@ -310,6 +312,25 @@ void ft_next_iov_cnt(struct ft_xcontrol *ctrl, size_t max_iov_cnt)
 		ctrl->iov_iter = 0;
 }
 
+int ft_next_ctx(struct ft_xcontrol *ctrl, uint64_t comp)
```

Comment:
I would modify this function to fully encapsulate getting an available context.  So we don't end up with this:
```
&ft_tx_ctrl.ctx[ft_tx_ctrl.curr_ctx]
```
at the call sites.  Convert to:
```
int ft_get_ctx(struct ft_xcontrol *ctrl, struct fi_context **ctx)
```
You can compare ctrl == &ft_tx_ctrl to determine if this is for a send or receive.  This will let us manage the contexts from a single location.